iroha_version/src/lib.rs

```diff
@@ -11,7 +11,38 @@ use serde::{Deserialize, Serialize};
 use std::ops::Range;
 
 pub mod error {
-    pub use iroha_error::*;
+    use iroha_derive::FromVariant;
+    use iroha_error::derive::Error;
+    use iroha_http_server::http::{HttpResponseError, StatusCode, HTTP_CODE_BAD_REQUEST};
+
+    #[derive(Error, Debug, FromVariant)]
+    pub enum Error {
+        #[error("This is not a versioned object. No version information found.")]
+        NotVersioned,
+        #[error("Can not encode unsupported version from json to scale.")]
+        UnsupportedJsonEncode,
+        #[error("Expected json object.")]
+        ExpectedJson,
+        #[error("Can not decode unsupported version from scale to json.")]
+        UnsupportedJsonDecode,
```

Comment:
```suggestion
        #[error("Can not serialize unsupported version from scale to json.")]
        UnsupportedScaleSerialize,
```

Reply:
Decode/encode, May be then serialize deserialize here?

Reply:
Yes, let me explain. There are two symmetric cases for this errors:
1. We have UnsupportedVersion::ScaleBytes and want to serialize it to json, which is impossible so we tell it.
2. We have UnsupportedVersion::Json and want to encode it to scale, which is impossible so we tell it.

So this is what I was trying to tell with naming and errors.

Reply:
UnsupportedVersion will only serialize into the format from which it came.